### PR TITLE
Adds new resource for azurerm_security_center_auto_provisioning

### LINKED ISSUE #8550: Support for Azure Security Center automatic provisioning
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure Security Center allows for the automatic provisioning of data collection agents on compute resources (virtual machines) in a given Azure subscription, [as described in the docs here](https://docs.microsoft.com/en-us/azure/security-center/security-center-enable-data-collection).

In summary this is a subscription wide setting and is a simple toggle "on" or "off", the enforcement of this setting is left to Azure Security Center

This setting can not be configured via Terraform today, but is needed by customers wishing to enforce certain policies and governance on subscriptions managed through Terraform.  
This setting is available in the Go SDK and the Azure ARM API.

### New or Affected Resource(s)

I propose a new resource under the *Security Center Resources* group, namely:

* azurerm_security_center_auto_provisioning

This resource would be nearly identical to the existing [`azurerm_security_center_subscription_pricing`](https://www.terraform.io/docs/providers/azurerm/r/security_center_subscription_pricing.html) resource, it would accept a string value for the auto provisioning setting, namely "on" or "off", [as described in the Azure API docs](https://docs.microsoft.com/en-us/rest/api/securitycenter/autoprovisioningsettings/get#autoprovision)

### Potential Terraform Configuration

```hcl
resource "azurerm_security_center_auto_provisioning" "example" {
  auto_provision = "On"
}
```

### References

* Existing issue: #2857
* https://docs.microsoft.com/en-us/azure/security-center/security-center-enable-data-collection
* https://docs.microsoft.com/en-us/rest/api/securitycenter/autoprovisioningsettings

### PR DESCRIPTION
**Pull request by benc-uk**
As per the proposal in issue #8550 this adds a new subscription level resource `azurerm_security_center_auto_provisioning` in order to support the auto provisioning feature of Security Center

Includes tests, which have been run and pass

Fixes #8550

### CODE REVIEW

File: azurerm/internal/services/securitycenter/azurerm_security_center_auto_provisioning.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	if properties := resp.AutoProvisioningSettingProperties; properties != nil {
+		d.Set("auto_provision", properties.AutoProvision)
+	}
+
+	return nil
+}
+
+func resourceArmSecurityCenterAutoProvisioningDelete(_ *schema.ResourceData, _ interface{}) error {
```
Comment (jackofallops) [inline azurerm/internal/services/securitycenter/azurerm_security_center_auto_provisioning.go:110]: Is there a "default" state that we should return this to? e.g. if it defaults to `Off`, and the user has it previously configured `On`, would the expectation from the user be that it is switched `Off` by deletion of the resource?

File: azurerm/internal/services/securitycenter/azurerm_security_center_auto_provisioning.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		return fmt.Errorf("Error creating/updating Security Center auto provisioning: %+v", err)
+	}
+
+	resp, err := client.Get(ctx, name)
+	if err != nil {
+		return fmt.Errorf("Error reading Security Center auto provisioning: %+v", err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Security Center auto provisioning ID is nil")
+	}
```
Comment (jackofallops) [inline azurerm/internal/services/securitycenter/azurerm_security_center_auto_provisioning.go:80]: Can we check for an empty value here too, setting an empty string in the state would be undesirable.
```suggestion
	if resp.ID == nil || *resp.ID == "" {
		return fmt.Errorf("Security Center Auto Provisioning ID missing")
	}
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @benc-uk 

Thanks for this PR, looking good so far, there's a few comments/questions below.  

Could you add the docs for this new resource too?

**Comment by benc-uk** [comment]
Hey @jackofallops all good suggestions, make all the changes plus a couple of other v minor improvements

**Review by jackofallops — APPROVED** [review]

**Comment by jackofallops** [comment]
Tests passing (1 fail unrelated)
image

**Pull request merged**